src/main/app/controller/lists/ContactTypesController.ts

```diff
@@ -0,0 +1,163 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {ContactType} from '../../../types/ContactType';
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+
+@autobind
+export class ContactTypesController {
+
+  editContactTypeUrl = 'lists/contact-type/';
+  deleteContactTypeUrl = 'lists/contact-type/';
+  deleteConfirmUrl = 'lists/contact-type/delete-confirm/';
+
+  getContactTypesError = 'An error occurred when retrieving the contact types.';
+  getContactTypeError = 'A problem occurred when retrieving the contact type.';
+  updateContactTypeError = 'A problem occurred when trying to save the contact type.';
+  contactTypeAlreadyExistsError = 'A contact type with the proposed name already exists. The name must be unique.';
+  deleteError = 'A problem occurred when trying to delete the contact type.';
+  contactTypeInUseError = 'You cannot delete this contact type at the moment, as one or more courts are dependent on it. ' +
+    'Please remove the contact type from the relevant courts first';
+  nameRequiredError = 'Name is required.';
+
+  public async getAll(req: AuthedRequest, res: Response): Promise<void> {
+    await this.renderAll(req, res);
+  }
+
+  public async getContactType(req: AuthedRequest, res: Response): Promise<void> {
+    const errors: { text: string }[] = [];
+    const id = req.params?.id;
+    let fatalError = false;
+
+    let contactType: ContactType = null;
+    if (id) {
+      await req.scope.cradle.api.getContactType(id)
+        .then((value: ContactType) => contactType = value )
+        .catch(() => {
+          errors.push({ text: this.getContactTypeError });
+          fatalError = true;
+        });
+    }
```

Comment:
What happens if id is blank, will this be checked on the njk file? Just double checking. Should we return an error message in this case?

Reply:
if id is blank error will be returned by api